### ISSUE TITLE
Problem: no typeahead template reuse

### ISSUE THREAD

**Issue by janat08** [Blockrazor/blockrazor#863]
Problem: typeahead is used in compare currencies and auctions, and is to be used for exchange add, has it's own CSS

Solution: @gazhayes should there be a reusable template for typeahead that takes params such as what to do on selection using currying to pass down parent template reference to change state and like what default query params it should and what method to use to initiate local collection.

**Comment by gazhayes** [comment]
Yes, this is worth doing (in my opinion). We use typeaheads very often and will probably be adding more.

**Comment by janat08** [comment]
removed "first" param used in curryEvent that was added for initializing from router params but seems to go unused.

**Comment by janat08** [comment]
implement id field prop

**Comment by janat08** [comment]
convert /auctions to it.
